realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -1001,12 +1012,48 @@ protected final boolean checkAndReportInvalidState() {
                 return false;
             }
 
+            // Handle Client Reset if it happened in any of the Realms.
+            // A Client Reset is a fatal error for the PermissionManager, so all current and
+            // future tasks will exit as soon as posible after this event happened and report it
+            // through the error callback. Only action a user can take is to close the
+            // PermissionManager and re-open it again Some data might be lost (like permission
+            // offers not yet processed). This is currently unavoidable.
+            // TODO: Eventually we might be able to recover the permission manager from this event
+            // but it will require some serious task management as we would need to do a full
+            // close, reschedule all tasks, and re-open behind users back. This is out of scope for
+            // now.
+            if (managementErrorHappened && managementError instanceof ClientResetRequiredError) {
+                ClientResetRequiredError cr = (ClientResetRequiredError) managementError;
+                permissionManager.managementRealm.close();
```

Comment:
I was about to say "Yes", because the PermissionManager is a singleton, but it is only a thread-singleton, so technically you could have a PermissionManager open on two different looper threads. I would estimate the chance of that ever happening to be virtually zero, but your right it is a concern.

In that case both PermissionManagers would receive the error callback and trigger this code. Both of them would "move" the file, which is unneassery work, but afaik it should be safe, since they would treat the error as fatal and not use the Realm anymore, so any data shouldn't be accessed on either thread. 

I'll create a unit test to verify that assumption.